pkg/syncer/fullsync.go

```diff
@@ -234,14 +235,26 @@ func fullSyncDeleteVolumes(ctx context.Context, volumeIDDeleteArray []cnstypes.C
 		return
 	}
 	// Verify if Volume is not in use by any other Cluster before removing CNS tag
+	vcVersion, err := metadataSyncer.volumeManager.GetVCVersion(ctx)
```

Comment:
How about just adding a new field to `metadatasyncer` called `vcVersion` which is initialized at startup? That way you can avoid calling this function all the time. 
Ofcourse that depends on the upgrade path; can a customer upgrade from 6.7u3 without restarting CSI?

Reply:
We also need a version check in the controller to publish capabilities depending on VC version, so metadata syncer is not just one usecase.

and actually, once the VC client is initialized this is just a static call to retrieve the VC client version from what it is inialzied with.